karapace/compatibility/jsonschema/checks.py

```diff
@@ -300,13 +326,14 @@ def compatibility_rec(
 def check_assertion_compatibility(
     reader_schema, writer_schema, assertion_check: AssertionCheck, location: List[str]
 ) -> SchemaCompatibilityResult:
-    result = SchemaCompatibilityResult.compatible()
+    result = SchemaCompatibilityResult(SchemaCompatibilityType.compatible)
 
     reader_value = reader_schema.get(assertion_check.keyword.value)
     writer_value = writer_schema.get(assertion_check.keyword.value)
 
     if introduced_constraint(reader_value, writer_value):
-        result.add_incompatibility(
+        add_incompatibility(
```

Comment:
Using a function for internal mutation instead of a method looks a bit odd, the version of compatibility result in the avro library no longer have this method?

Reply:
Yes, missing.
https://github.com/apache/avro/blob/release-1.11.0/lang/py/avro/compatibility.py#L81-L118